CMakeLists.txt

```diff
@@ -9,15 +9,29 @@ cmake_minimum_required(VERSION 3.10)
 # MacOS prior to 10.14 did not support aligned alloc which is used to implement
 # std::unique_ptr in the arm64 C++ standard library. x86_64 builds can override
 # this to 10.13.0 using -DCMAKE_OSX_DEPLOYMENT_TARGET="10.13.0" without issue.
-# This is done in the universal binary building script to build a binary that 
+# This is done in the universal binary building script to build a binary that
 # runs on 10.13 on x86_64 computers, while still containing an arm64 slice.
 
 set(CMAKE_OSX_DEPLOYMENT_TARGET "10.14.0" CACHE STRING "")
 
 set(CMAKE_USER_MAKE_RULES_OVERRIDE "CMake/FlagsOverride.cmake")
 
+# Optionally enable these polcies so older versions of cmake don't break.
+# we only need them for MSVC
+if(POLICY CMP0091)
+  cmake_policy(SET CMP0091 NEW) # MSVC runtime library flags are selected by an abstraction.
+  cmake_policy(SET CMP0092 NEW) # MSVC warning flags are not in CMAKE_{C,CXX}_FLAGS by default.
+endif()
+
 project(dolphin-emu)
 
+if (MSVC)
+  set(CMAKE_C_STANDARD 99)
+  set(CMAKE_CXX_STANDARD 23)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
afaiui, this causes cmake to pass `Za` for C++ sources(?), something that msbuild setup is currently not doing. It's probably better to leave out for now

Reply:
As far as I can tell, this option does absolutely nothing on MSVC.

Reply:
Remove it then?